contracts/modules/Checkpoint/EtherDividendCheckpoint.sol

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * @dev Issuer can push dividends to provided addresses
+     * @param _dividendIndex Dividend to push
+     * @param _payees Addresses to which to push the dividend
+     */
+    function pushDividendPaymentToAddresses(uint256 _dividendIndex, address[] _payees) public onlyOwner validDividendIndex(_dividendIndex) {
+        Dividend storage dividend = dividends[_dividendIndex];
+        for (uint256 i = 0; i < _payees.length; i++) {
+            if (!dividend.claimed[_payees[i]]) {
+                _payDividend(_payees[i], dividend, _dividendIndex);
+            }
+        }
+    }
+
+    /**
+     * @dev Issuer can push dividends using the investor list from the security token
+     * @param _dividendIndex Dividend to push
+     * @param _start Index in investor list at which to start pushing dividends
+     * @param _iterations Number of addresses to push dividends for
+     */
+    function pushDividendPayment(uint256 _dividendIndex, uint256 _start, uint256 _iterations) public onlyOwner validDividendIndex(_dividendIndex) {
+        Dividend storage dividend = dividends[_dividendIndex];
+        uint256 numberInvestors = ISecurityToken(securityToken).getInvestorsLength();
+        for (uint256 i = _start; i < Math.min256(numberInvestors, _start.add(_iterations)); i++) {
+            address payee = ISecurityToken(securityToken).investors(i);
+            if (!dividend.claimed[payee]) {
+                _payDividend(payee, dividend, _dividendIndex);
+            }
+        }
+    }
+
+    /**
+     * @dev Investors can pull their own dividends
+     * @param _dividendIndex Dividend to pull
+     */
+    function pullDividendPayment(uint256 _dividendIndex) public validDividendIndex(_dividendIndex)
+    {
+        Dividend storage dividend = dividends[_dividendIndex];
+        require(dividend.claimed[msg.sender] == false);
+        _payDividend(msg.sender, dividend, _dividendIndex);
+    }
+
+    function _payDividend(address _payee, Dividend storage _dividend, uint256 _dividendIndex) internal {
+        uint256 claim = calculateDividend(_dividendIndex, _payee);
+        _dividend.claimed[_payee] = true;
+        _dividend.claimedAmount = claim.add(_dividend.claimedAmount);
+        if (claim > 0) {
+            _payee.transfer(claim);
+            emit EtherDividendClaimed(_payee, _dividendIndex, claim);
+        }
+    }
+
+    /**
+     * @dev Issuer can reclaim remaining unclaimed dividend amounts, for expired dividends
+     * @param _dividendIndex Dividend to reclaim
+     */
+    function reclaimDividend(uint256 _dividendIndex) public onlyOwner {
+        require(_dividendIndex < dividends.length, "Incorrect dividend index");
+        require(now >= dividends[_dividendIndex].expiry, "Dividend expiry is in the future");
+        require(!dividends[_dividendIndex].reclaimed, "Dividend already claimed");
+        dividends[_dividendIndex].reclaimed = true;
+        Dividend storage dividend = dividends[_dividendIndex];
```

Comment:
Since you are already doing `Dividend storage dividend = dividends[_dividendIndex];`
you could put the line above down here and change it to be `dividend.reclaimed = true;`

Reply:
Done